builtin/common/misc_helpers.lua

```diff
@@ -447,21 +447,35 @@ end
 
 
 --------------------------------------------------------------------------------
-function core.string_to_area(value)
+function core.string_to_area_internal(value, is_relative_to)
```

Comment:
This function is backwards-compatible and adds functionalities modders might also want to make use of.
I think this function name can be kept, and `lua_api.txt` extended to document the additional parameter.